### ISSUE TITLE
Same dataset(180 GB), why training 1 epoch on 2 nodes(16 gpus) takes around 10 minutes, training on 4 nodes(32 gpus) only take 33 seconds?

### ISSUE THREAD

**Issue by GravityZL** [tensorflow/tensorflow#47160]
# System Information:
TensorFlow version (use command below): 2.1.1
Python version: 3.7
mpi/openmpi: 4.0.3-gnu-9.2.0
compiler/gnu/: 9.2.0
CUDA version: 10.1.243
GPU model and memory:  Tesla V100-SXM2-32GB

# Description:

I tested the code with multiple nodes in a cluster, and the training time for one epoch has sudden decrease when use 4 nodes, and I have no idea why it happens. The codes tries to read from 343 csv files(around 180 gb in total), and each csv file has 14406 records, each record has 4860 features, and 8 labels to be predicted. The 8 labels are numbers, thus it is a regression problem.

Here is the code: 

# Code: 
```
import time
import pathlib
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras.layers import Conv2D, AveragePooling2D, MaxPool2D, Flatten, Dense, BatchNormalization, Dropout, Activation
from tensorflow.keras.optimizers import SGD, RMSprop
import sys
import json
import os
import socket
import datetime
import math
import pathlib

tf.random.set_seed(22)

hostlist=sys.argv[1].split(',')
current_host=socket.gethostname()
index=hostlist.index(current_host)

simsInFile=14406

verbose = index < 1
nodes=[]

for host in hostlist:
  nodes.append(host + ':2001')

os.environ['TF_CONFIG'] = json.dumps({
  'cluster': {
    'worker': nodes
  },
  'task': {'type': 'worker', 'index': index}
})
print(os.environ['TF_CONFIG'])

strategy = tf.distribute.experimental.MultiWorkerMirroredStrategy(tf.distribute.experimental.CollectiveCommunication.AUTO)

AUTOTUNE = tf.data.experimental.AUTOTUNE

per_worker_batch_size = 512
num_workers = strategy.num_replicas_in_sync
global_batch_size = per_worker_batch_size * num_workers

num_epochs = 1

def read_csv(line):
  fields = tf.io.decode_csv(line, record_defaults=[tf.constant([], dtype=tf.float32)]*4869, field_delim=",")
  label = tf.stack(fields[4861:4869])
  label = tf.dtypes.cast(label, tf.dtypes.float32)
  label = tf.reshape(label,[8])

  features = tf.stack(fields[1:4861])
  features = tf.dtypes.cast(features, tf.dtypes.float32)
  features = tf.reshape(features,[10,486])

  return features, label

def create_train_dataset(ds_files_, batch_size=global_batch_size, nr_epochs=num_epochs, buffer_size=10000):
  dataset = ds_files_.interleave(lambda x:tf.data.TextLineDataset(x).skip(1))
  dataset = dataset.map(read_csv, num_parallel_calls=AUTOTUNE)
  dataset = dataset.cache()
  dataset = dataset.shuffle(buffer_size=buffer_size)
  dataset = dataset.repeat()
  dataset = dataset.batch(batch_size)
  dataset = dataset.prefetch(1)
  return dataset

learning_rate = 1e-3
total_epoch = 1
ds_files = tf.data.Dataset.list_files(xxx/*.csv')
ds_files = ds_files.shard(num_workers, index)
train_dataset = create_train_dataset(ds_files)

def build_model():
  return tf.keras.models.Sequential([
    tf.keras.layers.Conv1D(64,3,input_shape=(10, 486)),
    tf.keras.layers.Conv1D(64,3),
    tf.keras.layers.Dropout(0.3),
    tf.keras.layers.MaxPooling1D(pool_size=2),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(100, activation='relu'),
    tf.keras.layers.Dense(8)
  ])

with strategy.scope():
  model = build_model()
  model.compile(optimizer=RMSprop(learning_rate), loss='mse')
  start_time = datetime.datetime.now()
  steps_epoch = 9651//num_workers
  history = model.fit(train_dataset,steps_per_epoch = steps_epoch,epochs = num_epochs, verbose = 1)
  print("Training Finished!")

os.environ.pop('TF_CONFIG', None)
```

**Comment by GravityZL** [comment]
I found the reason:  the training time should exponentially decay as the number of GPUs increases
